rest/tests/test_rest_authorization.py

```diff
@@ -0,0 +1,476 @@
+        self.logout()
+        self.login(self.curator_user.username)
+
+        self.create_repository(expected_status=HTTP_403_FORBIDDEN)
+
+        self.create_repository({
+            'name': 'newname',
+            'description': 'description',
+        }, expected_status=HTTP_403_FORBIDDEN)
+
+        # switch to user with no permissions for repo
+        self.logout()
+        self.login(self.user_norepo.username)
+
+        self.create_repository({
+            'name': 'newname',
+            'description': 'description',
+        }, expected_status=HTTP_403_FORBIDDEN)
+
+        # as anonymous
+        self.logout()
+        self.create_repository({
+            'name': 'newname',
+            'description': 'description',
+        }, expected_status=HTTP_403_FORBIDDEN)
+
+    def test_repository_patch_put(self):
+        """Test updating repository"""
+        # switch to add_repo_user
+        self.logout()
+        self.login(self.add_repo_user.username)
+
+        input_dict = {
+            'name': self.repo.name,
+            'description': 'new description'
+        }
+        self.patch_repository(self.repo.slug, input_dict,
+                              expected_status=HTTP_403_FORBIDDEN)
+        self.put_repository(self.repo.slug, input_dict,
+                            expected_status=HTTP_403_FORBIDDEN)
+
+        # switch to curator_user
+        self.logout()
+        self.login(self.curator_user.username)
+        self.patch_repository(self.repo.slug, input_dict,
+                              expected_status=HTTP_405_METHOD_NOT_ALLOWED)
+        self.put_repository(self.repo.slug, input_dict,
+                            expected_status=HTTP_405_METHOD_NOT_ALLOWED)
+
+        # switch to user with no permissions for repo
+        self.logout()
+        self.login(self.user_norepo.username)
+        self.patch_repository(self.repo.slug, input_dict,
+                              expected_status=HTTP_403_FORBIDDEN)
+        self.put_repository(self.repo.slug, input_dict,
+                            expected_status=HTTP_403_FORBIDDEN)
+
+        # as anonymous
+        self.logout()
+        self.patch_repository(self.repo.slug, input_dict,
+                              expected_status=HTTP_403_FORBIDDEN)
+        self.put_repository(self.repo.slug, input_dict,
+                            expected_status=HTTP_403_FORBIDDEN)
+
+    def test_repository_delete(self):
+        """Test deleting repository"""
+        # switch to add_repo_user
+        self.logout()
+        self.login(self.add_repo_user.username)
+
+        self.delete_repository(self.repo.slug,
+                               expected_status=HTTP_403_FORBIDDEN)
+
+        self.logout()
+        self.login(self.curator_user.username)
+
+        # curator does have view_repo but nobody is allowed
+        # to delete repositories
+        self.delete_repository(self.repo.slug,
+                               expected_status=HTTP_405_METHOD_NOT_ALLOWED)
```

Comment:
I understand it's intentional, but why is this a 405 instead of 403, if the action isn't allowed?